parser-typechecker/src/Unison/Server/Backend.hs

```diff
@@ -323,7 +327,8 @@ typeListEntry codebase r n = do
       pure $ case decl of
         Just (Left _) -> Ability
         _             -> Data
-    _ -> pure Data
+    -- IO is the only built-in ability
+    _ -> pure (if r == builtInIoAbility then Ability else Data)
```

Comment:
```suggestion
    _ -> pure (if r == Type.builtinIORef then Ability else Data)
```

Might need to `import qualified Unison.Type as Type` if that doesn't already exist in this module.